models/DataObject/Service.php

```diff
@@ -1513,6 +1513,12 @@ public static function enrichLayoutDefinition(&$layout, $object = null, $context
         if (method_exists($layout, 'getChildren')) {
             $children = $layout->getChildren();
             if (is_array($children)) {
+                // Send information when we have block or similar element
+                if ($layout instanceof \Pimcore\Model\DataObject\ClassDefinition\Data && empty($context['subContainerType'])) {
```

Comment:
How about moving the assignment outside `getChildren` scope and add condition for Block specific type next to localize fields and classificationstore? as it will eventually call `enrichLayoutDefinition` on children as well.
```php
    $context['ownerName'] = 'localizedfields';
} elseif ($layout instanceof DataObject\ClassDefinition\Data\Block && isset($context['containerType'])) {    
    $context['subContainerKey'] = $layout->getName();
    $context['subContainerType'] = $layout->getFieldtype();
}
```